src/status_im/wallet/accounts/core.cljs

```diff
@@ -13,78 +13,191 @@
+    ""
+    (derive-and-store-account constants/path-default-wallet hashed-password :seed))))
+
+(re-frame/reg-fx
+ ::import-account-private-key
+ (fn [{:keys [private-key hashed-password]}]
+   (status/multiaccount-import-private-key
+    (security/unmask private-key)
+    (store-account constants/path-default-wallet hashed-password :key))))
 
 (fx/defn generate-new-account
-  {:events [:wallet.accounts/generate-new-account]}
-  [{:keys [db]} password]
+  [{:keys [db]} hashed-password]
   (let [wallet-root-address (get-in db [:multiaccount :wallet-root-address])
         path-num            (inc (get-in db [:multiaccount :latest-derived-path]))]
-    (when-not (get-in db [:add-account :step])
-      {:db                (assoc-in db [:add-account :step] :generating)
-       ::generate-account {:derivation-info (if wallet-root-address
-                                              ;; Use the walllet-root-address for stored on disk keys
-                                              ;; This needs to be the RELATIVE path to the key used to derive
-                                              {:path    (str "m/" path-num)
-                                               :address wallet-root-address}
-                                              ;; Fallback on the master account for keycards, use the absolute path
-                                              {:path    (str constants/path-wallet-root "/" path-num)
-                                               :address (get-in db [:multiaccount :address])})
-                           :path-num        path-num
-                           :hashed-password (ethereum/sha3 password)}})))
-
-(fx/defn generate-new-account-error
-  {:events [::generate-new-account-error]}
-  [{:keys [db]} password]
-  {:db (assoc db
-              :add-account
-              {:error (i18n/label :t/add-account-incorrect-password)})})
+    {:db                (assoc-in db [:add-account :step] :generating)
+     ::generate-account {:derivation-info (if wallet-root-address
+                                            ;; Use the walllet-root-address for stored on disk keys
+                                            ;; This needs to be the RELATIVE path to the key used to derive
+                                            {:path    (str "m/" path-num)
+                                             :address wallet-root-address}
+                                            ;; Fallback on the master account for keycards, use the absolute path
+                                            {:path    (str constants/path-wallet-root "/" path-num)
+                                             :address (get-in db [:multiaccount :address])})
+                         :hashed-password hashed-password}}))
+
+(fx/defn import-new-account-seed
+  [{:keys [db]} passphrase hashed-password]
+  {:db                   (assoc-in db [:add-account :step] :generating)
+   :validate-mnemonic   [(security/safe-unmask-data passphrase)
```

Comment:
keep the namespaced keywords and import via alias, if there is a circular dep it just means that the effect is not in the right place from a code organization pov

Reply:
fixed